aws/rust-runtime/aws-config/src/imds/client.rs

```diff
@@ -673,67 +658,71 @@ pub(crate) mod test {
             .unwrap()
     }
 
-    pub(crate) fn imds_response(body: &'static str) -> http::Response<&'static str> {
-        http::Response::builder().status(200).body(body).unwrap()
+    pub(crate) fn imds_response(body: &'static str) -> HttpResponse {
+        http::Response::builder()
+            .status(200)
+            .body(SdkBody::from(body))
+            .unwrap()
     }
 
-    pub(crate) fn make_client<T>(conn: &TestConnection<T>) -> super::Client
-    where
-        SdkBody: From<T>,
-        T: Send + 'static,
-    {
+    pub(crate) fn make_client(http_client: &EventClient) -> super::Client {
```

Comment:
Yeah, I haven't been liking this name so much, but I wasn't able to think of a better one. `StaticReplayClient` seems reasonable.

Reply:
They key here is differentiating it from DVR's ReplayingClient, and I think the Static somewhat accomplishes that. It's already behind `test-util`, so that it's test should be clear.